### PR TITLE
DOR-79 Backend > Agregar portada de Libros

### PR DESCRIPTION
**Pull request by Andrei-R1**
https://isc-ads.atlassian.net/browse/DOR-79

### CODE REVIEW

**Comment by Andrei-R1** [comment]
Im still unable to retrieve errors, I dont know if the problems come from the image server, but currently it only has to get the server error messages and cut the workflow, at the moment if there is an error in the image, it get ignored and because it can be a null value and will register the other parts of the data

**Comment by wmartzh** [comment]
Remember thar is another API so the errors are http errors, for example send another file that is ni an image

File: src/controllers/book.controller.ts [hunk 1]
```diff
@@ -2,21 +2,29 @@ import { Request, Response } from "express";
...
    * It validates the request body with the RegisterBookSchema, then calls the bookService.createBook function, and finally sends the response
    * @param {Request | any} req
    * @param {Response} res
    */
-  async createBook(req: Request | any, res: Response) {
+  async createBook(req: Request | any, res: Response): Promise<void> {
     try {
-      const data = await RegisterBookSchema.validateAsync(req.body);
+      const data = await RegisterBookSchema.validateAsync(req.body)
+      if (req.file) {
```
Comment (wmartzh) [inline src/controllers/book.controller.ts]: This logic should be inside the service, controller has single responsibility to handle request

File: src/controllers/book.controller.ts [hunk 2]
```diff
@@ -2,21 +2,29 @@ import { Request, Response } from "express";
...
+        const picture = await imageService.uploadImage(req.file);
+        data.picture = picture;
+      } 
       const result = await bookService.createBook(data);
       this.responseHandler(res, { message: `Book ${result.title} created successfully`}, 200);
     } catch (error: any) {
       if (error.code && error.code === "P2002") {
         this.errorHandler(res, { error: "Book was already registered" });
+      } else if (error.response && error.response.status === 400) {
```
Comment (wmartzh) [inline src/controllers/book.controller.ts]: When you connect a external service, you should't mix errors,  `imageService` should handle those errors

Comment (Andrei-R1) [reply src/controllers/book.controller.ts]: Done

**Comment by Andrei-R1** [comment]
Now it works and throws all errors with its message correctly, also has a specific created but without image message, and returns the image server error

**Pull request merged**